File: c/gpt/p108.c
#include <stdio.h>
#include <stdlib.h>

/* p108.c
   Run: ./p108 <username>
   Use popen to run `getent passwd <username>` and print its output.
   NOTE: This is a simple example and does not sanitize input.
*/
int main(int argc, char **argv) {
    if (argc < 2) { fprintf(stderr, "usage: %s username\n", argv[0]); return 1; }
    char cmd[256];
    snprintf(cmd, sizeof(cmd), "getent passwd %s", argv[1]);
    FILE *fp = popen(cmd, "r");
    if (!fp) { perror("popen"); return 1; }
    char line[1024];
    while (fgets(line, sizeof(line), fp)) fputs(line, stdout);
    pclose(fp);
    return 0;
}

File: java/gpt/p108.java
import java.io.*;

// p108.java - run `getent passwd <username>` and print output
public class p108 {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) { System.err.println("usage: p108 <username>"); System.exit(1); }
        String user = args[0];
        ProcessBuilder pb = new ProcessBuilder("getent", "passwd", user);
        pb.redirectErrorStream(true);
        Process p = pb.start();
        try (BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
            String line;
            while ((line = r.readLine()) != null) System.out.println(line);
        }
        int rc = p.waitFor();
        System.exit(rc);
    }
}
